sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1114 @@
+          related facts.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, symbols, I
+        >>> x = symbols('x')
+        >>> ask(Q.negative(x), Q.real(x) & ~Q.positive(x) & ~Q.zero(x))
+        True
+        >>> ask(Q.negative(-1))
+        True
+        >>> ask(Q.nonnegative(I))
+        False
+        >>> ask(~Q.negative(I))
+        True
+
+        """
+        return Predicate('negative')
+
+    @memoize_property(predicate_storage)
+    def zero(self):
+        """
+        Zero number predicate.
+
+        ``ask(Q.zero(x))`` is true iff the value of ``x`` is zero.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, oo
+        >>> ask(Q.zero(0))
+        True
+        >>> ask(Q.zero(1/oo))
+        True
+        >>> ask(Q.zero(0*oo))
+        False
+        >>> ask(Q.zero(1))
+        False
+
+        """
+        return Predicate('zero')
+
+    @memoize_property(predicate_storage)
+    def nonzero(self):
+        """
+        Nonzero real number predicate.
+
+        ``ask(Q.nonzero(x))`` is true iff ``x`` is real and ``x`` is not zero.  Note in
+        particular that ``Q.nonzero(x)`` is false if ``x`` is not real.  Use
+        ``~Q.zero(x)`` if you want the negation of being zero without any real
+        assumptions.
+
+        A few important facts about nonzero numbers:
+
+        - ``Q.nonzero`` is logically equivalent to ``Q.positive | Q.negative``.
+
+        - See the documentation of ``Q.real`` for more information about
+          related facts.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, symbols, I
+        >>> x = symbols('x')
+        >>> print(ask(Q.nonzero(x), ~Q.zero(x)))
+        None
+        >>> ask(Q.nonzero(x), Q.positive(x))
+        True
+        >>> ask(Q.nonzero(x), Q.zero(x))
+        False
+        >>> ask(Q.nonzero(0))
+        False
+        >>> ask(Q.nonzero(I)) # doctest: +SKIP
+        False
+        >>> ask(~Q.zero(I))
+        True
+
+        """
+        return Predicate('nonzero')
+
+    @memoize_property(predicate_storage)
+    def nonpositive(self):
+        """
+        Nonpositive real number predicate.
+
+        ``ask(Q.nonpositive(x))`` is true iff ``x`` belongs to the set of
+        negative numbers including zero.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, I
+        >>> ask(Q.nonpositive(-1))
+        True
+        >>> ask(Q.nonpositive(0))
+        True
+        >>> ask(Q.nonpositive(1))
+        False
+        >>> ask(Q.nonpositive(I))
+        False
+        >>> ask(Q.nonpositive(-I))
+        False
+
+        """
+        return Predicate('nonpositive')
+
+    @memoize_property(predicate_storage)
+    def nonnegative(self):
+        """
+        Nonnegative real number predicate.
+
+        ``ask(Q.nonnegative(x))`` is true iff ``x`` belongs to the set of
+        positive numbers including zero.
```

Comment:
I wonder if we should move/copy the notes about nonpositive and nonnegative from the `Q.positive` and `Q.negative` docstrings to these ones. It's also probably worth repeating that these are all finite. Having an interval here as with the others would be a good thing.